enterprise/cloud.oracle/src/org/netbeans/modules/cloud/oracle/assets/k8s/PortForwards.java

```diff
@@ -0,0 +1,235 @@
+     * This method retrieves the Pod's ports and forwards them to the local
+     * machine. If any port is unavailable, the operation is halted for that
+     * port and a notification is displayed.</p>
+     *
+     * @param podItem The {@code PodItem} representing the Pod for which port
+     * forwarding should start.
+     */
+    public void startPortForward(PodItem podItem) {
+        if (activePortForwards.containsKey(podItem.getName())) {
+            NotificationUtils.showMessage("Port forwarding already active for: " + podItem.getName());
+            return;
+        }
+        CountDownLatch latch = new CountDownLatch(1);
+        ProgressHandle handle = ProgressHandle.createHandle(Bundle.Forwarding(podItem.getName()), () -> {
+            latch.countDown();
+            return true;
+        });
+        handle.start();
+        CompletableFuture<Void> future = CompletableFuture.runAsync(() -> {
+            KubernetesUtils.runWithClient(podItem.getCluster(), client -> {
+                Pod pod = client.pods().inNamespace(podItem.getNamespace()).withName(podItem.getName()).get();
+
+                if (pod == null) {
+                    System.out.println("Pod not found: " + podItem.getName());
+                    return;
+                }
+                List<Integer> ports = pod.getSpec().getContainers().stream()
+                        .flatMap(container -> container.getPorts().stream())
+                        .map(port -> port.getContainerPort())
+                        .collect(Collectors.toList());
+                List<PortForwardItem> forwardItems = new ArrayList<>();
+                List<PortForward> forwards = new ArrayList<>();
+                if (ports.isEmpty()) {
+                    NotificationUtils.showMessage("No ports found for pod: " + podItem.getName());
+                    return;
+                }
+                try {
+                    for (Integer port : ports) {
+                        if (!isPortAvailable(port)) {
+                            NotificationUtils.showErrorMessage(Bundle.PortNotFree(port));
+                            break;
+                        }
+                        LocalPortForward fwd = client.pods()
+                                .inNamespace(pod.getMetadata().getNamespace())
+                                .withName(pod.getMetadata().getName())
+                                .portForward(port, port);
+                        forwards.add(fwd);
+                        forwardItems.add(new PortForwardItem(podItem, fwd.getLocalPort(), port));
+
+                        NotificationUtils.showMessage(Bundle.ForwardingPorts(port.toString(), podItem.getName()));
+                    }
+                    latches.put(podItem.getName(), latch);
+                    List<PortForwardItem> oldValue = activePortForwards.put(podItem.getName(), forwardItems);
+                    firePropertyChange(podItem, oldValue, forwardItems);
+                    latch.await();
```

Comment:
Consider using other approach than CDL here. Each Pod will block its thread until it is requested to close. Ideally the request to close should execute the closing procedure, eventually asynchronously.

Please add `@OnStop` handler that closes port forwards upon NB shutdown. Eventually load *active* port forwards from the k8s after startup, if possible.